conn.go

```diff
@@ -127,7 +127,6 @@ func (c *Conn) listenResponse(id uint32) (*Header, error) {
 	defer func() {
 		c.Lock()
 		close(c.listeners[id])
-		delete(c.listeners, id)
```

Comment:
It's probably better to make sure that map accesses are protected, or look at doing this a different way.